.github/workflows/e2e_test.yaml

```diff
@@ -1,470 +1,32 @@
-        if: always() && (matrix.event.name == 'issues' || matrix.event.name == 'schedule')
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{matrix.event.name}}-workflow-logs.zip
-          path: ${{matrix.event.name}}-workflow-logs.zip
-          if-no-files-found: ignore
-
-      - name: Show Firewall Rules
-        run: |
-          juju ssh ${{ steps.runner-name.outputs.name }}/0 sudo nft list ruleset
-
-      - name: Clean Up (Workflow Dispatch and Push)
-        if: always() && (matrix.event.name == 'workflow_dispatch' || matrix.event.name == 'push')
-        env:
-          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          gh api \
-          --method DELETE \
-          -H "Accept: application/vnd.github+json" \
-          -H "X-GitHub-Api-Version: 2022-11-28" \
-          "/repos/${{ secrets.E2E_TESTING_REPO }}/git/refs/heads/${{ steps.runner-name.outputs.name }}"
-          echo "Deleted ref ${{ steps.runner-name.outputs.name }}"
-
-      - name: Clean Up (Issues, Schedule)
-        if: always() && (matrix.event.name == 'issues' || matrix.event.name == 'schedule')
-        env:
-          GH_TOKEN: ${{ secrets.E2E_TESTING_TOKEN }}
-        run: |
-          set +e
-
-          gh api \
-          --method DELETE \
-          -H "Accept: application/vnd.github+json" \
-          -H "X-GitHub-Api-Version: 2022-11-28" \
-          "/repos/${{ secrets.E2E_TESTING_REPO }}/git/refs/heads/${{ steps.runner-name.outputs.name }}" \
-          && echo "Deleted ref ${{ steps.runner-name.outputs.name }}"
-
-          TESTING_REPO=${{ secrets.E2E_TESTING_TOKEN_ORG }}/${{ steps.runner-name.outputs.name }}
-
-          set -e
-
-          gh api \
-          --method DELETE \
-          -H "Accept: application/vnd.github+json" \
-          -H "X-GitHub-Api-Version: 2022-11-28" \
-          "/repos/${TESTING_REPO}"
-
-          echo "Deleted repo ${TESTING_REPO}"
-
-  e2e-test:
-    name: End-to-End Test
-    needs: [build-charm, run-id]
-    uses: ./.github/workflows/e2e_test_run.yaml
+  # test option values defined at test/conftest.py are passed on via repository secret
+  # INTEGRATION_TEST_ARGS to operator-workflows automatically.
+  openstack-integration-end-to-end-test:
+    name: end-to-end test using private-endpoint
+    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
+    secrets: inherit
     with:
-      runner-tag: "pr-${{ needs.run-id.outputs.run-id }}${{ github.run_attempt}}"
-      runner-virt-type: "lxd"
+      juju-channel: 3.2/stable
+      pre-run-script: scripts/setup-lxd.sh
+      provider: lxd
+      test-tox-env: integration-juju3.2
+      modules: '["test_e2e"]'
+      extra-arguments: "-m openstack --openstack-flavor-name=builder-cpu4-ram8-disk50 --http-proxy=http://squid.internal:3128 --https-proxy=http://squid.internal:3128 --no-proxy=keystone.ps6.canonical.com,glance.ps6.canonical.com,nova.ps6.canonical.com,neutron.ps6.canonical.com"
```

Comment:
This is information about internal infrastructure, should it really be public or should it be placed in a secret?